src/aggregator/client/conn.go

```diff
@@ -178,10 +213,26 @@ func (c *connection) connectWithLock() error {
 		c.metrics.setKeepAliveError.Inc(1)
 	}
 
-	if c.conn != nil {
-		c.conn.Close() // nolint: errcheck
+	if c.tcpConn != nil {
+		c.tcpConn.Close() // nolint: errcheck
+	}
+	c.tcpConn = tcpConn
+
+	var writer connWriter
+	switch c.compressType {
+	case NoCompressType:
+		writer = newRawConnWriter(tcpConn)
+	case SnappyCompressType:
+		c.snappyWriter.Reset(tcpConn)
+		writer = c.snappyWriter
+	case FlateCompressType:
+		c.flateWriter.Reset(tcpConn)
+		writer = c.flateWriter
```

Comment:
Looks like we do this and the the connReader init in a couple of different places; might be worth adding a function that takes in a `CompressType` and a `*net.TCPConn` and returns the compression-wrapped connection?